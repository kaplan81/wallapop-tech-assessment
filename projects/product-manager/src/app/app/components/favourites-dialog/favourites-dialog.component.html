<div class="favourites-dialog">
  <ul class="favourites-dialog__favourites">
    <li
      class="favourites-dialog__favourites__favourite"
      *ngFor="let favouriteEntity of favouriteEntities$ | async"
    >
      <mng-product
        isInDialog
        [product]="favouriteEntity"
        (closeDialog)="closeDialog()"
      ></mng-product>
    </li>
  </ul>
  <div class="favourites-dialog__actions" mat-dialog-actions>
    <button class="favourites-dialog__actions__close-btn" mat-button mat-dialog-close>Close</button>
  </div>
</div>
