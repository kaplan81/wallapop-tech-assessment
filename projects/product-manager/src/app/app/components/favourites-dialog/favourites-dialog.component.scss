.favourites-dialog {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 40px;
  &__favourites {
    margin-top: 30px;
    &__viewport {
      height: 60vh;
      margin-bottom: 5px;
      &__favourite {
        height: 117px;
        &:not(:last-child) {
          margin-bottom: 20px;
        }
      }
    }
  }
  &__actions {
    position: absolute;
    display: flex;
    justify-content: center;
    width: 100%;
    bottom: 0;
    right: 0;
    padding-bottom: 30px;
    &__close-btn {
      width: 200px;
      height: 45px;
      border: 2px solid #000;
    }
  }
}
