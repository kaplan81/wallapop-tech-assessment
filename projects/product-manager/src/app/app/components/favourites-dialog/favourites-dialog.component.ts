import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [],
  templateUrl: './favourites-dialog.component.html',
  selector: 'mng-favourites-dialog',
  standalone: true,
  styleUrls: ['./favourites-dialog.component.scss'],
})
export class FavouritesDialogComponent {}
