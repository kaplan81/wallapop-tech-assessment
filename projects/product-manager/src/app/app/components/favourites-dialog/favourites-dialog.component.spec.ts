import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialogRef } from '@angular/material/dialog';
import { FavouritesDialogComponent } from './favourites-dialog.component';

describe('FavouritesDialogComponent', () => {
  let fixture: ComponentFixture<FavouritesDialogComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [FavouritesDialogComponent],
      providers: [{ provide: MatDialogRef, useValue: {} }],
    }).compileComponents();
    fixture = TestBed.createComponent(FavouritesDialogComponent);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
