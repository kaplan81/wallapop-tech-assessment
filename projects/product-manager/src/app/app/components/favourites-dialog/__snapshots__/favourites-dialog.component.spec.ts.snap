// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesDialogComponent should match snapshot 1`] = `
<mng-favourites-dialog
  __ngContext__="0"
  dialogRef={[Function Object]}
  favouriteEntities$={[Function Observable]}
>
  <div
    class="favourites-dialog"
  >
    <ul
      class="favourites-dialog__favourites"
    >
      <cdk-virtual-scroll-viewport
        class="cdk-virtual-scroll-viewport favourites-dialog__favourites__viewport cdk-virtual-scrollable cdk-virtual-scroll-orientation-vertical"
        itemsize="50"
      >
        <div
          class="cdk-virtual-scroll-content-wrapper"
        >
        </div>
        <div
          class="cdk-virtual-scroll-spacer"
        />
      </cdk-virtual-scroll-viewport>
    </ul>
    <div
      class="mat-mdc-dialog-actions mdc-dialog__actions favourites-dialog__actions"
      mat-dialog-actions=""
    >
      <button
        class="favourites-dialog__actions__close-btn mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        mat-button=""
        mat-dialog-close=""
        type="button"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
          Close
        </span>
        <span
          class="mat-mdc-focus-indicator"
        />
        <span
          class="mat-ripple mat-mdc-button-ripple"
          matripple=""
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
    </div>
  </div>
</mng-favourites-dialog>
`;
